sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -148,32 +148,53 @@ public void sendEvent(Event event) {
      * @param eventBuilder {@link EventBuilder} to send to Sentry.
      */
     public void sendEvent(EventBuilder eventBuilder) {
-        if (!Util.isNullOrEmpty(release)) {
+        Event event = buildEvent(eventBuilder);
+        sendEvent(event);
+    }
+
+    /**
+     * Adds the SentryClient data into the builder without overriding existent data and builds the event.
+     * @param eventBuilder The event builder to potentially modify and to build the event.
+     * @return The event.
+     */
+    Event buildEvent(EventBuilder eventBuilder) {
+        Event workingEvent = eventBuilder.getEvent();
+
+        if (!Util.isNullOrEmpty(release) && workingEvent.getRelease() == null) {
```

Comment:
Can't you reuse `isNullOrEmpty` in both cases? Unless empty release in event is valid usecase.
Same goes for the rest of checks.

Reply:
Personally I'd use `null` check everywhere. If we need to drop empty strings I'd do it in a single place (i.e right before sending on some event validation layer) but on each single operation over every string doing null or empty kind of sucks. And also all the `.trim()` calls everywhere. I don't want to change `Util.isNullOrEmpty` into a simple `null` check because that is a backward incompatible change though.